From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: domicile CI <ci@domicile.invalid>
Date: Wed, 30 Sep 2026 00:00:00 +0000
Subject: [PATCH] domicile: a shell is told an address to open

A DomicileOpenUrlEvent carries an address `domicile open-url` asked the
desktop to open -- what BROWSER runs inside it -- to the shell page.

The interface is a new file under src/; what cannot be a new file is a
name in a generated-bindings list.
---
 third_party/blink/renderer/bindings/generated_in_modules.gni | 4 ++++
 third_party/blink/renderer/bindings/idl_in_modules.gni       | 2 ++
 2 files changed, 6 insertions(+)

diff --git a/third_party/blink/renderer/bindings/generated_in_modules.gni b/third_party/blink/renderer/bindings/generated_in_modules.gni
index d77509e050..87afea43e1 100644
--- a/third_party/blink/renderer/bindings/generated_in_modules.gni
+++ b/third_party/blink/renderer/bindings/generated_in_modules.gni
@@ -341,6 +341,8 @@ generated_dictionary_sources_in_modules = [
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_domicile_theme_event_init.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_domicile_tray_event_init.cc",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_domicile_tray_event_init.h",
+  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_domicile_open_url_event_init.cc",
+  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_domicile_open_url_event_init.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_domicile_notifications_event_init.cc",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_domicile_notifications_event_init.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_domicile_locked_event_init.cc",
@@ -2442,6 +2444,8 @@ generated_interface_sources_in_modules = [
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_domicile_tray_event.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_domicile_tray_item.cc",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_domicile_tray_item.h",
+  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_domicile_open_url_event.cc",
+  "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_domicile_open_url_event.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_domicile_notification.cc",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_domicile_notification.h",
   "$root_gen_dir/third_party/blink/renderer/bindings/modules/v8/v8_domicile_notification_action.cc",
diff --git a/third_party/blink/renderer/bindings/idl_in_modules.gni b/third_party/blink/renderer/bindings/idl_in_modules.gni
index 448e5d11ce..c6cfd4dc5f 100644
--- a/third_party/blink/renderer/bindings/idl_in_modules.gni
+++ b/third_party/blink/renderer/bindings/idl_in_modules.gni
@@ -317,6 +317,8 @@ static_idl_files_in_modules = [
   "//third_party/blink/renderer/modules/domicile/domicile_tray_event.idl",
   "//third_party/blink/renderer/modules/domicile/domicile_tray_event_init.idl",
   "//third_party/blink/renderer/modules/domicile/domicile_tray_item.idl",
+  "//third_party/blink/renderer/modules/domicile/domicile_open_url_event.idl",
+  "//third_party/blink/renderer/modules/domicile/domicile_open_url_event_init.idl",
   "//third_party/blink/renderer/modules/domicile/domicile_notification.idl",
   "//third_party/blink/renderer/modules/domicile/domicile_notification_action.idl",
   "//third_party/blink/renderer/modules/domicile/domicile_notifications_event.idl",
-- 
2.43.0

